components/remote_settings/src/storage.rs

```diff
@@ -140,38 +141,31 @@ impl Storage {
         }
     }
 
-    /// Set the list of records stored in the database, clearing out any previously stored records
-    pub fn set_records(
+    // TODO
+    pub fn set_collection_content(
         &mut self,
         collection_url: &str,
         records: &Vec<RemoteSettingsRecord>,
+        timestamp: u64,
+        metadata: CollectionMetadata,
```

Comment:
As you pointed out, we're not doing the right thing when syncing with both new and old records.  Should we handle that here?  Based on the other changes, I think `records` could be renamed `changes` and we could implement the merging here while deleting/inserting things into the DB.

Reply:
Let's work on this in a separate PR, and I can rebase this work on top 👌

Reply:
See #6526